pkcs7/src/cryptographic_message_syntax2004.rs

```diff
@@ -0,0 +1,378 @@
+///   crls \[1\] IMPLICIT RevocationInfoChoices OPTIONAL,
+///   signerInfos SignerInfos }
+#[derive(Clone, Eq, PartialEq, Sequence)]
+pub struct SignedData<'a> {
+    ///   version CMSVersion,
+    pub version: u8,
+    ///   digestAlgorithms DigestAlgorithmIdentifiers,
+    pub digest_algorithms: DigestAlgorithmIdentifiers<'a>,
+    ///   encapContentInfo EncapsulatedContentInfo,
+    pub encap_content_info: EncapsulatedContentInfo<'a>,
+    // Using Any as a means of deferring most of the decoding of the certificates (will still need
+    // to call to_vec on the resulting Any to restore tag and length values).
+    ///   certificates \[0\] IMPLICIT CertificateSet OPTIONAL,
+    #[asn1(context_specific = "0", tag_mode = "IMPLICIT", optional = "true")]
+    pub certificates: Option<alloc::vec::Vec<Any<'a>>>,
+    ///   crls \[1\] IMPLICIT RevocationInfoChoices OPTIONAL,
+    #[asn1(context_specific = "1", tag_mode = "IMPLICIT", optional = "true")]
+    pub crls: Option<alloc::vec::Vec<Any<'a>>>,
+    ///   signerInfos SignerInfos }
+    pub signer_infos: SetOfVec<SignerInfo<'a>>,
+}
+
+/// DigestAlgorithmIdentifiers ::= SET OF DigestAlgorithmIdentifier
+pub type DigestAlgorithmIdentifiers<'a> = SetOfVec<ObjectIdentifier>;
+
+/// EncapsulatedContentInfo ::= SEQUENCE {
+///   eContentType ContentType,
+///   eContent \[0\] EXPLICIT OCTET STRING OPTIONAL }
+#[derive(Clone, Eq, PartialEq, Sequence)]
+pub struct EncapsulatedContentInfo<'a> {
+    ///   eContentType ContentType,
+    pub econtent_type: ObjectIdentifier,
+    ///   eContent \[0\] EXPLICIT OCTET STRING OPTIONAL }
+    #[asn1(context_specific = "0", tag_mode = "EXPLICIT", optional = "true")]
+    pub econtent: Option<OctetString<'a>>,
+}
+
+// ContentType ::= OBJECT IDENTIFIER
+
+/// SignerInfo ::= SEQUENCE {
+///   version CMSVersion,
+///   sid SignerIdentifier,
+///   digestAlgorithm DigestAlgorithmIdentifier,
+///   signedAttrs \[0\] IMPLICIT SignedAttributes OPTIONAL,
+///   signatureAlgorithm SignatureAlgorithmIdentifier,
+///   signature SignatureValue,
+///   unsignedAttrs \[1\] IMPLICIT UnsignedAttributes OPTIONAL }
+#[derive(Clone, Eq, PartialEq, PartialOrd, Sequence)]
+pub struct SignerInfo<'a> {
+    ///   version CMSVersion,
+    pub version: u8,
+    ///   sid SignerIdentifier,
+    pub sid: SignerIdentifier<'a>,
+    ///   digestAlgorithm DigestAlgorithmIdentifier,
+    pub digest_algorithm: AlgorithmIdentifier<'a>,
+    ///   signedAttrs \[0\] IMPLICIT SignedAttributes OPTIONAL,
+    pub signed_attrs: SignedAttributes<'a>,
+    ///   signatureAlgorithm SignatureAlgorithmIdentifier,
+    pub signature_algorithm: AlgorithmIdentifier<'a>,
+    ///   signature SignatureValue,
+    pub signature: BitString<'a>,
+    ///   unsignedAttrs \[1\] IMPLICIT UnsignedAttributes OPTIONAL }
+    pub unsigned_attrs: UnsignedAttributes<'a>,
+}
+impl ValueOrd for SignerInfo<'_> {
+    fn value_cmp(&self, _other: &Self) -> der::Result<Ordering> {
+        todo!()
```

Comment:
Pull requests must not contain outstanding `todo!()` panics.

Reply:
This is known to be incomplete.

Reply:
That means nothing to me. This code will still panic at runtime if this method is called. Runtime panics are a NO GO because they provide opportunities for DoS attacks.